src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php

```diff
@@ -0,0 +1,199 @@
+use PrestaShopBundle\Form\Admin\Type\SwitchType;
+use PrestaShopBundle\Translation\TranslatorAwareTrait;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\BirthdayType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\EmailType;
+use Symfony\Component\Form\Extension\Core\Type\IntegerType;
+use Symfony\Component\Form\Extension\Core\Type\NumberType;
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * Type is used to created form for customer add/edit actions
+ */
+class CustomerType extends AbstractType
+{
+    use TranslatorAwareTrait;
+
+    /**
+     * @var array
+     */
+    private $genderChoices;
+
+    /**
+     * @var array
+     */
+    private $groupChoices;
+
+    /**
+     * @var bool
+     */
+    private $isB2bFeatureEnabled;
+
+    /**
+     * @var array
+     */
+    private $riskChoices;
+
+    /**
+     * @var bool
+     */
+    private $isPartnerOffersEnabled;
+
+    /**
+     * @param array $genderChoices
+     * @param array $groupChoices
+     * @param array $riskChoices
+     * @param bool $isB2bFeatureEnabled
+     * @param bool $isPartnerOffersEnabled
+     */
+    public function __construct(
+        array $genderChoices,
+        array $groupChoices,
+        array $riskChoices,
+        $isB2bFeatureEnabled,
+        $isPartnerOffersEnabled
+    ) {
+        $this->genderChoices = $genderChoices;
+        $this->groupChoices = $groupChoices;
+        $this->isB2bFeatureEnabled = $isB2bFeatureEnabled;
+        $this->riskChoices = $riskChoices;
+        $this->isPartnerOffersEnabled = $isPartnerOffersEnabled;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('gender_id', ChoiceType::class, [
+                'choices' => $this->genderChoices,
+                'multiple' => false,
+                'expanded' => true,
+                'required' => false,
+                'placeholder' => null,
+            ])
+            ->add('first_name', TextType::class, [
+                'constraints' => [
+                    new NotBlank([
+                        'message' => $this->trans('This field cannot be empty', [], 'Admin.Notifications.Error'),
+                    ]),
+                    new Length([
+                        'max' => FirstName::MAX_LENGTH,
+                        'maxMessage' => $this->trans(
+                            'This field cannot be longer than %limit% characters',
+                            ['%limit%' => FirstName::MAX_LENGTH],
+                            'Admin.Notifications.Error'
```

Comment:
@LouiseBonnard @matks since i want to use validator that is available in symfony i faced an issue when there are no appropriate error messages available.
Thus i introduced new error messages, can you check and confirm? If its ok, ill use these messages (or introduce other when needed) for other fields as well.

Reply:
@LouiseBonnard  It is ok for me !